coral/src/services/router-utils/route-utils.tsx

```diff
@@ -0,0 +1,280 @@
+  redirectUnauthorized: Routes;
+  children?: RouteObject[];
+};
+
+type SuperadminRoute = {
+  path: Routes;
+  element: JSX.Element;
+  redirect: string;
+  children?: RouteObject[];
+};
+
+function isEnvironmentsTabEnum(value: unknown): value is EnvironmentsTabEnum {
+  if (isString(value)) {
+    return Object.prototype.hasOwnProperty.call(
+      ENVIRONMENT_TAB_ID_INTO_PATH,
+      value
+    );
+  }
+  return false;
+}
+function isTopicsOverviewTabEnum(
+  value: unknown
+): value is TopicOverviewTabEnum {
+  if (isString(value)) {
+    return Object.prototype.hasOwnProperty.call(
+      TOPIC_OVERVIEW_TAB_ID_INTO_PATH,
+      value
+    );
+  }
+  return false;
+}
+
+function isConnectorsOverviewTabEnum(
+  value: unknown
+): value is ConnectorOverviewTabEnum {
+  if (isString(value)) {
+    return Object.prototype.hasOwnProperty.call(
+      CONNECTOR_OVERVIEW_TAB_ID_INTO_PATH,
+      value
+    );
+  }
+  return false;
+}
+
+function isRequestsTabEnum(value: unknown): value is RequestsTabEnum {
+  if (isString(value)) {
+    return Object.prototype.hasOwnProperty.call(
+      REQUESTS_TAB_ID_INTO_PATH,
+      value
+    );
+  }
+  return false;
+}
+
+function isApprovalsTabEnum(value: unknown): value is ApprovalsTabEnum {
+  if (isString(value)) {
+    return Object.prototype.hasOwnProperty.call(
+      APPROVALS_TAB_ID_INTO_PATH,
+      value
+    );
+  }
+  return false;
+}
+function createRouteBehindFeatureFlag({
+  path,
+  element,
+  featureFlag,
+  redirectRouteWithoutFeatureFlag,
+  children,
+}: FeatureFlagRouteArgs) {
+  return {
+    path: path,
+    // if FEATURE_FLAG_<name> is not set to
+    // true, we're not even rendering the page but
+    // redirect user directly to /topics
+    element: isFeatureFlagActive(featureFlag) ? (
+      element
+    ) : (
+      <Navigate to={redirectRouteWithoutFeatureFlag} />
+    ),
+    ...(children && { children }),
+  };
+}
+
+const PrivateRoute = ({
+  children,
+  permission,
+  redirectUnauthorized,
+}: {
+  children: React.ReactNode;
+  permission: Permission;
+  redirectUnauthorized: Routes;
+}) => {
+  const { permissions } = useAuthContext();
+  const toast = useToast();
+  const isAuthorized = permissions[permission];
+
+  useEffect(() => {
+    if (!isAuthorized) {
+      toast({
+        message: `Not authorized`,
+        position: "bottom-left",
+        variant: "danger",
+      });
+
+      return;
+    }
+  }, []);
+
+  return isAuthorized ? (
+    children
+  ) : (
+    <Navigate to={redirectUnauthorized} replace />
+  );
+};
+
+function createPrivateRoute({
+  path,
+  element,
+  permission,
+  redirectUnauthorized,
+  children,
+}: PrivateRouteArgs) {
+  return {
+    path: path,
+    element: (
+      <PrivateRoute
+        permission={permission}
+        redirectUnauthorized={redirectUnauthorized}
+      >
+        {element}
+      </PrivateRoute>
+    ),
+    ...(children && { children }),
+  };
+}
+
+const SuperadminRoute = ({
+  children,
+  redirect,
```

Comment:
Should this be named `redirectSuperadmin` so that we keep it as explicit as the other `redirect` prop on other routes? Would also make logic like this less prone to break if we introcued another route component using `redirect` as a prop:

```tsx        
  redirect !== undefined ? (
            <SuperadminRoute
              redirect={redirect}
              showNotFound={redirectMap[routePath]?.showNotFound}
              removeChildren={redirectMap[routePath]?.removeChildren}
            >
              {route.element}
            </SuperadminRoute>
          ) : (
            route.element
          ),
```

Reply:
Yes, that's good, will change the name!

Reply:
https://github.com/Aiven-Open/klaw/pull/2303/commits/42d20082974ee18f58a10430d4de5cfdd93fa37f